src/app/OperationalSessionSetup.cpp

```diff
@@ -292,12 +294,17 @@ void OperationalSessionSetup::UpdateDeviceData(const Transport::PeerAddress & ad
     // Do not touch `this` instance anymore; it has been destroyed in DequeueConnectionCallbacks.
 }
 
-CHIP_ERROR OperationalSessionSetup::EstablishConnection(const ReliableMessageProtocolConfig & config)
+CHIP_ERROR OperationalSessionSetup::EstablishConnection(const ResolveResult & result)
 {
+    auto & config = result.mrpRemoteConfig;
 #if INET_CONFIG_ENABLE_TCP_ENDPOINT
-    // TODO: Combine LargePayload flag with DNS-SD advertisements from peer.
-    // Issue #32348.
-    if (mTransportPayloadCapability == TransportPayloadCapability::kLargePayload)
+    if (mTransportPayloadCapability == TransportPayloadCapability::kLargePayload && !result.supportsTcpServer)
+    {
+        // we should not set the large payload while the TCP support is not enabled
+        ChipLogError(Discovery, "LargePayload support is enabled while TCP server support is not enabled");
```

Comment:
Please change the error log to something like "LargePayload session requested but peer does not support TCP server" to clarify that one came down from the local app, while the other is a parameter of the peer.

Reply:
done